JavaScript/JavaScript.sublime-syntax

```diff
@@ -15,6 +15,22 @@ variables:
   func_lookahead: '\s*\b(async\s+)?function\b'
   arrow_func_lookahead: '\s*(\basync\s*)?([_$[:alpha:]][_$[:alnum:]]*|\(([^()]|\([^()]*\))*\))\s*=>'
 
+  method_name: >-
+    (?x)(?:
+      {{identifier}}
+      | '(?:[^\\']|\\.)*'
+      | "(?:[^\\"]|\\.)*"
+      | \[ {{identifier}} (?:\.{{identifier}})* \]
+    )
+
+  line_continuation_lookahead: >-
+    (?x)
+    (?! \+\+ | -- )
+    (?=
+      != |
+      [ -+*/% ><= &|^ \[( ;,.:? ]
+    )
```

Comment:
It has to do with JavaScript's automatic semicolon insertion algorithm.

JavaScript expression statements end in a semicolon. However, the parser effectively makes semicolons optional in many cases by automatically inserting them under some circumstances. This means that a newline may represent the end of a statement depending on the contents of the following line. This lookahead examines the following line to see whether the expression continues.